packages/react/src/SelectPanel/SelectPanel.tsx

```diff
@@ -328,40 +340,106 @@ export function SelectPanel({
+      resetSort()
+    }
+  }, [filterValue])
+
+  useEffect(() => {
+    if (open) {
+      resetSort()
+    }
+  }, [open])
+
+  useEffect(() => {
+    if (prevItems.length == 0 && items.length > 0) {
+      resetSort()
+    }
+  }, [items])
+
   const itemsToRender = useMemo(() => {
-    return items.map(item => {
-      const isItemSelected = isMultiSelectVariant(selected) ? doesItemsIncludeItem(selected, item) : selected === item
-
-      return {
-        ...item,
-        role: 'option',
-        selected: 'selected' in item && item.selected === undefined ? undefined : isItemSelected,
-        onAction: (itemFromAction, event) => {
-          item.onAction?.(itemFromAction, event)
-
-          if (event.defaultPrevented) {
-            return
-          }
+    return items
+      .map(item => {
+        const isItemSelected = isMultiSelectVariant(selected) ? doesItemsIncludeItem(selected, item) : selected === item
+
+        return {
+          ...item,
+          role: 'option',
+          selected: 'selected' in item && item.selected === undefined ? undefined : isItemSelected,
+          onAction: (itemFromAction, event) => {
+            item.onAction?.(itemFromAction, event)
+
+            if (event.defaultPrevented) {
+              return
+            }
 
-          if (isMultiSelectVariant(selected)) {
-            const otherSelectedItems = selected.filter(selectedItem => !areItemsEqual(selectedItem, item))
-            const newSelectedItems = doesItemsIncludeItem(selected, item)
-              ? otherSelectedItems
-              : [...otherSelectedItems, item]
+            if (isMultiSelectVariant(selected)) {
+              const otherSelectedItems = selected.filter(selectedItem => !areItemsEqual(selectedItem, item))
+              const newSelectedItems = doesItemsIncludeItem(selected, item)
+                ? otherSelectedItems
+                : [...otherSelectedItems, item]
 
-            const multiSelectOnChange = onSelectedChange as SelectPanelMultiSelection['onSelectedChange']
-            multiSelectOnChange(newSelectedItems)
-            return
-          }
+              const multiSelectOnChange = onSelectedChange as SelectPanelMultiSelection['onSelectedChange']
+              multiSelectOnChange(newSelectedItems)
+              return
+            }
 
-          // single select
-          const singleSelectOnChange = onSelectedChange as SelectPanelSingleSelection['onSelectedChange']
-          singleSelectOnChange(item === selected ? undefined : item)
-          onClose('selection')
-        },
-      } as ItemProps
-    })
-  }, [onClose, onSelectedChange, items, selected])
+            // single select
+            const singleSelectOnChange = onSelectedChange as SelectPanelSingleSelection['onSelectedChange']
+            singleSelectOnChange(item === selected ? undefined : item)
+            onClose('selection')
+          },
+        } as ItemProps
+      })
+      .sort((itemA, itemB) => {
```

Comment:
I'm coding this as the default behavior. Wondering if we should allow consumer to opt-out of this behavior or provide their custom sort function 👀 
cc: @emilybrick

Reply:
I feel strongly that we should let consumers opt-out by providing their own sort function. I think DataTable has a nice API for handling this: https://github.com/primer/react/blob/main/packages/react/src/DataTable/column.ts#L59